modules/internal/String.chpl

```diff
@@ -2347,8 +2357,8 @@ module String {
 
   pragma "no doc"
   inline proc param string.byte(param i: int) param : uint(8) {
-    if i < 1 || i > this.numBytes then
-      compilerError("index out of bounds of string: " + i:string);
+    if i < 0 || i > this.numBytes-1 then
+      compilerError("index " + i:string + " out of bounds for string with " + this.size:string + " bytes");
```

Comment:
`this.size` returns number of codepoints, not bytes. You should use `this.numBytes` instead.

Reply:
Having gone through all the other changes, I did another round for checking the error messages, and I think all the others use the right thing except for this one.

Reply:
Good catch!  I'd attempted to double-check these as well yesterday, but apparently didn't do a very good job of it...

Reply:
@e-kayrakli: https://github.com/chapel-lang/chapel/pull/14671/commits/c6a1525b05088632373d386f984c8317bb08edc0 fixes this and adds a few tests to lock the behavior in (previous tests had not).  Would you review that specific commit for me?

Reply:
Looks good!